Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -759,8 +759,17 @@ def network_shutdown(self):
             self.threadpool.cancel_all_pending_tasks()
             self.threadpool = None
 
+    def save_download_pstate_callback(self, resume_data, infohash, pstate):
+        pstate.set('state', 'engineresumedata', resume_data)
+        self.save_download_pstate(infohash, pstate)
```

Comment:
I guess you need a check for that then. Note that LMC extends TaskManager (`class TriblerLaunchMany(TaskManager):`), so if it solves your problem you can let the taskmanager cancel the deferreds for you on shutdown. (note that this may not be the right place to cancel them, you should check this)